### ISSUE TITLE
[v5.0.0-beta.1] Hanging request

### ISSUE THREAD

**Issue by bileslav** [amphp/http-client#302]
```console
subroot@BH:~/racoon-2/test$ php -v
PHP 8.1.5 (cli) (built: Apr 22 2022 04:56:05) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.5, Copyright (c) Zend Technologies
	with Zend OPcache v8.1.5, Copyright (c), by Zend Technologies
```
```php
// hang.php

ini_set('error_reporting', E_ALL);
ini_set('error_log', null);

require 'vendor/autoload.php';

$client = Amp\Http\Client\HttpClientBuilder::buildDefault();
$request = new Amp\Http\Client\Request('https://www.youtube.com/watch?v=Yw6u6YkTgQ4');

$request->setProtocolVersions(['1.1']);

for ($i = 0; $i < 100; $i++) {
	$cancellation = new Amp\TimeoutCancellation(10); // even that doesn't help

	if ($client->request($request, $cancellation)->getStatus() !== 200) {
		throw new LogicException();
	}

	echo $i, ' ';
}

echo "\n";
```
Here's an output of the "successful" test - the execution just freezes after "24" (may be another number):
```console
subroot@BH:~/racoon-2/test$ php hang.php
0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24
```
- not only do timeouts from `Request` not work, the optional `TimeoutCancellation` also doesn't work
- the issue may take several executions to appear
- not sure if it appears when using non-HTTP/1.1
- tested with `StreamSelectDriver` and `UvDriver` - same result

**Comment by trowski** [comment]
I can reproduce this, but only with the HTTP/1.x driver. It freezes after a varying number of requests. Will need to investigate.

**Comment by bileslav** [comment]
It seems that I found the culprit and a potential solution. The bottom line is that a `while` at [Http1Connection.php:290](https://github.com/amphp/http-client/blob/b3203552c84dc950372ed421bdb3d8d8c931751f/src/Connection/Http1Connection.php#L290) goes to infinity under some condition. I don't really understand what's going on in this file, so my solution is probably wrong, but I hope it will help anyway: [Http1Connection.php (fixed)](https://pastebin.com/8i8ZYdmC)

**Comment by trowski** [comment]
Thanks for providing your fix, it pointed me directly at what was wrong so I could write a more complete fix. The fix has now been tagged under `v5.0.0-beta.2`, so please give it a try. Feel free to reopen this issue if you still have issues.
